src/bcf/match_variants.rs

```diff
@@ -99,19 +118,27 @@ pub fn match_variants(
                 .map(|a| {
                     if let Some(range) = index.range(chrom, pos) {
                         for v in Itertools::flatten(range.map(|(_, idx_vars)| idx_vars)) {
-                            if let Some(id) = var.matches(v, a, max_dist, max_len_diff) {
-                                return id as i32;
+                            if let Some(id) = var.matches(v, a, max_dist, max_len_diff)? {
+                                return Ok(id as i32);
                             }
                         }
                     }
-                    -1
+                    Ok(-1)
                 })
-                .collect_vec();
+                .collect::<errors::Result<Vec<i32>>>()?;
 
-            rec.push_info_integer(b"MATCHING", &matching)?;
-            outbcf.write(&rec)?;
+            rec.push_info_integer(b"MATCHING", &matching)
+                .context(errors::BCFTagWriteError {
+                    data: format!("{:?}", matching),
+                    fd: String::from("MATCHING"),
+                })?;
+            outbcf.write(&rec).context(errors::BCFWriteError {
+                record: format!("{:?}", rec),
```

Comment:
Can the formatting be automated inside the error? Why is it needed here, and not above?

Reply:
As discussed records have been removed from write errors output messages.